research-app/src/App.vue

```diff
@@ -461,6 +503,34 @@ export default class App extends WWTAwareComponent {
         this.onMessage(event.data);
       }
     }, false);
+
+    // Handling key presses
+    this._keyboardControlSettings = new KeyboardControlSettings({});
+    this._keyListener = function(e: KeyboardEvent) {
+      const kcs = this._keyboardControlSettings;
+      if (kcs == null || kcs == undefined) { return; }
+      const movementAmount = kcs.moveAmount;
+      const tiltAmount = kcs.tiltAmount;
+      if (e.key === kcs.zoomIn) {
```

Comment:
If there are going to be multiple keys that trigger the same action (e.g. I/J/K/L and arrow keys for panning), then this part of things might need to be restructured. But this is one of those situations where there's no 100% elegant way to do things, so adding fields like `kcs.zoomInA` and `kcs.zoomInB` or whatever would be totally fine.

Reply:
I guess another option would be to make the keybinding fields in KeyboardControlSettings into arrays of strings, and then the check could be something like `kcs.zoomIn.contains(e.key)`